UnitsNet.Tests/QuantityTypeConverterTest.cs

```diff
@@ -0,0 +1,348 @@
+            ITypeDescriptorContext context = new TypeDescriptorContext("SomeMemberName", new Attribute[] { });
+
+            Assert.Throws<NotSupportedException>(() => unitsNETTypeConverter.ConvertFrom(context, culture, ""));
+        }
+
+        [Fact]
+        public void ConvertFrom_GiveWrongQuantity()
+        {
+            UnitsNetTypeConverter<Length> unitsNETTypeConverter = new UnitsNetTypeConverter<Length>();
+            ITypeDescriptorContext context = new TypeDescriptorContext("SomeMemberName", new Attribute[] { });
+
+            Assert.Throws<ArgumentException>(() => unitsNETTypeConverter.ConvertFrom(context, culture, "1m^2"));
+        }
+
+        [Theory]
+        [InlineData(typeof(Length))]
+        [InlineData(typeof(IQuantity))]
+        [InlineData(typeof(object))]
+        public void ConvertTo_GiveWrongType_ThrowException(Type value)
+        {
+            UnitsNetTypeConverter<Length> unitsNETTypeConverter = new UnitsNetTypeConverter<Length>();
+            ITypeDescriptorContext context = new TypeDescriptorContext("SomeMemberName", new Attribute[] { });
+            Length length = Length.FromMeters(1);
+
+            Assert.Throws<NotSupportedException>(() => unitsNETTypeConverter.ConvertTo(length, value));
+        }
+
+        [Theory]
+        [InlineData(typeof(string))]
+        public void ConvertTo_GiveRigthType_DoConvertion(Type value)
+        {
+            UnitsNetTypeConverter<Length> unitsNETTypeConverter = new UnitsNetTypeConverter<Length>();
+            ITypeDescriptorContext context = new TypeDescriptorContext("SomeMemberName", new Attribute[] { });
+            Length length = Length.FromMeters(1);
+
+            Assert.Equal("1 m", unitsNETTypeConverter.ConvertTo(length, value));
+        }
+
+        [Fact]
+        public void ConvertTo_GiveSomeValues_ConvertToString()
+        {
+            UnitsNetTypeConverter<Length> unitsNETTypeConverter = new UnitsNetTypeConverter<Length>();
+            ITypeDescriptorContext context = new TypeDescriptorContext("SomeMemberName", new Attribute[] { });
+            Length length = Length.FromMeters(1);
+
+            Assert.Equal("1 m", unitsNETTypeConverter.ConvertTo(length, typeof(string)));
+            Assert.Equal("1 m", unitsNETTypeConverter.ConvertTo(context, culture, length, typeof(string)));
+
+            context = new TypeDescriptorContext("SomeMemberName", new Attribute[]
+            {
+                new DefaultUnitAttribute(Units.LengthUnit.Centimeter),
+                new ConvertToUnitAttribute(Units.LengthUnit.Millimeter)
+            });
+
+            Assert.Equal("1 m", unitsNETTypeConverter.ConvertTo(length, typeof(string)));
+            Assert.Equal("1 m", unitsNETTypeConverter.ConvertTo(context, culture, length, typeof(string)));
+
+            context = new TypeDescriptorContext("SomeMemberName", new Attribute[]
+            {
+                new DefaultUnitAttribute(Units.LengthUnit.Centimeter),
```

Comment:
Is DefaultUnitAttribute used in these ConvertTo tests? I thought it controlled the default unit in ConvertFrom cases, for strings with only numbers?
Same with ConvertToUnitAttribute, is that used?

Reply:
No they are not used.
This test is only to make sure that the y do not influence the displayed unit.

Reply:
I think it's best to test only one thing, or at least minimize the variables of the test case. If you want to test that the other attributes don't have an effect in a particular setup then that can be a separate test method I think.